macros/utils/cross_db_utils/timeadd.sql

```diff
@@ -9,7 +9,13 @@
 {% endmacro %}
 
 {% macro bigquery__edr_timeadd(date_part, number, timestamp_expression) %}
-    timestamp_add({{ elementary.edr_cast_as_timestamp(timestamp_expression) }}, INTERVAL {{ number }} {{ date_part }})
+    {%- if date_part | lower in ['second', 'minute', 'hour', 'day'] %}
+       timestamp_add({{ elementary.edr_cast_as_timestamp(timestamp_expression) }}, INTERVAL {{ number }} {{ date_part }})
+    {%- elif date_part | lower in ['week', 'month', 'quarter', 'year'] %}
+       date_add({{ elementary.edr_cast_as_date(timestamp_expression) }}, INTERVAL {{ number }} {{ date_part }})
```

Comment:
For BigQuery we cast as `date` and then add a week / month - would we lose the information on the hour-of-day ? Also - is this by intentional?

Reply:
1. We have no choice, adding day/week/month to timestamp is not supported in BQ
2. We truncate days to 00:00:00 across the code anyway

Reply:
Got it